tests/tests/fee_tests.cpp

```diff
@@ -213,6 +213,126 @@ BOOST_AUTO_TEST_CASE(asset_claim_fees_test)
+        transfer( committee_account, bob_id, _core(1000) );
+
+        enable_fees();
+
+        auto claim_pool = [&]( const account_id_type& issuer, const asset_id_type& asset_to_claim,
+                              const asset& amount_to_fund, const asset_object& fee_asset  )
+        {
+            asset_claim_pool_operation claim_op;
+            claim_op.issuer = issuer;
+            claim_op.asset_id = asset_to_claim;
+            claim_op.amount_to_claim = amount_to_fund;
+
+            signed_transaction tx;
+            tx.operations.push_back( claim_op );
+            db.current_fee_schedule().set_fee( tx.operations.back(), fee_asset.options.core_exchange_rate );
+            set_expiration( db, tx );
+            sign( tx, alice_private_key );
+            PUSH_TX( db, tx );
+
+        };
+
+        auto claim_pool_proposal = [&]( const account_id_type& issuer, const asset_id_type& asset_to_claim,
+                                        const asset& amount_to_fund, const asset_object& fee_asset  )
+        {
+            asset_claim_pool_operation claim_op;
+            claim_op.issuer = issuer;
+            claim_op.asset_id = asset_to_claim;
+            claim_op.amount_to_claim = amount_to_fund;
+
+            const auto proposal_create_fees = fees.get<proposal_create_operation>();
+            proposal_create_operation prop;
+            prop.fee_paying_account = alice_id;
+            prop.proposed_ops.emplace_back( claim_op );
+            prop.expiration_time =  db.head_block_time() + fc::days(1);
+            prop.fee = asset( proposal_create_fees.fee + proposal_create_fees.price_per_kbyte );
+
+            signed_transaction tx;
+            tx.operations.push_back( prop );
+            db.current_fee_schedule().set_fee( tx.operations.back(), fee_asset.options.core_exchange_rate );
+            set_expiration( db, tx );
+            sign( tx, alice_private_key );
+            PUSH_TX( db, tx );
+
+        };
+
+        const asset_object& core_asset = asset_id_type()(db);
+
+        // deposit 100 BTS to the fee pool of ALICEUSD asset
+        fund_fee_pool( alice, aliceusd, _core(100).amount );
+
+        // Unable to claim pool before the hardfork
+        GRAPHENE_REQUIRE_THROW( claim_pool( alice_id, aliceusd.id, _core(1), core_asset), fc::exception );
+        GRAPHENE_REQUIRE_THROW( claim_pool_proposal( alice_id, aliceusd.id, _core(1), core_asset), fc::exception );
+
+        // Fast forward to hard fork date
+        generate_blocks( HARDFORK_CORE_188_TIME );
+
+        // can't claim pool because it is empty
+        GRAPHENE_REQUIRE_THROW( claim_pool( alice_id, alicecoin.id, _core(1), core_asset), fc::exception );
+
+        // deposit 300 BTS to the fee pool of ALICECOIN asset
+        fund_fee_pool( alice, alicecoin, _core(300).amount );
+
+        // Test amount of CORE in fee pools
+        BOOST_CHECK( alicecoin.dynamic_asset_data_id(db).fee_pool == _core(300).amount );
```

Comment:
Better change this check (and checks for equality below) to
`BOOST_CHECK_EQUAL( alicecoin.dynamic_asset_data_id(db).fee_pool.value, 300)`

Reply:
interestingly, the tests fail this way, but not when using `_core().amount` .. do I maybe need to cast to uint64, or so?